pkg/aws/validations/iam_helpers.go

```diff
@@ -0,0 +1,60 @@
+package validations
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/openshift/rosa/pkg/aws/tags"
+	semver "github.com/hashicorp/go-version"
+)
+
+func GetRoleName(prefix string, role string) string {
+	name := fmt.Sprintf("%s-%s-Role", prefix, role)
+	if len(name) > 64 {
+		name = name[0:64]
+	}
+	return name
+}
+
+func IsManagedRole(roleTags []*iam.Tag) bool {
+    for _, tag := range roleTags {
+        if aws.StringValue(tag.Key) == tags.ManagedPolicies && aws.StringValue(tag.Value) == "true" {
+            return true
+        }
+    }
+
+    return false
+}
+
+func HasCompatibleVersionTags(iamTags []*iam.Tag, version string) (bool, error) {
+	if len(iamTags) == 0 {
+		return false, nil
+	}
+	for _, tag := range iamTags {
+		if aws.StringValue(tag.Key) == tags.OpenShiftVersion {
+			if version == aws.StringValue(tag.Value) {
+				return true, nil
+			}
+			wantedVersion, err := semver.NewVersion(version)
+			if err != nil {
+				return false, err
+			}
```

Comment:
Can we check this outside the loop. Is there a reason we are doing this ever iteration?

Reply:
I can move that logic to an external function but I don't see a way of checking it outside of the for loop without making it messy

Reply:
oh my mistake I thought you meant both current and wanted version